### ISSUE TITLE
Unable to change widget location and use global widget

### ISSUE THREAD

**Issue by SimpleByDesign** [sojimaxi/grav-plugin-widget#7]
Hi,

I am having trouble setting up my own theme with widgets.

I have the following in my theme config:

```
widget:
  areas: [topbar,header,headerleft,headermiddle,headerright,abovebody,sidebar,belowbody,footer]
```
 This works fine, when adding widgets to a page or under the global site configuration these are listed as expected.

The theme was working correctly when I used the following:

`{{ widget( 'footer', false ) }}`

However I realised I wanted a global widget to appear here so I altered the theme code to:

`{{ widget( 'footer') }}`

However I now get the following error:

`Twig_Error_Runtime thrown with message "An exception has been thrown during the rendering of a template ("Undefined property: stdClass::$widget")."`

I realised that the widget name was the same as the block name in the main theme template, so I thought I will rename the widget areas, so that the theme config file now contains the following:

```
widget:
  areas: [widgettopbar,widgetheader,widgetheaderleft,widgetheadermiddle,widgetheaderright,widgetabovebody,widgetsidebar,widgetbelowbody,widgetfooter]
```

However the widget plugin never appears to up date to these new options. I have uninstalled the plugin, deleted my theme then cleared all the cache and then installed again. But for some reason, when I re-install my theme (with new widget locations) and the widget plugin; it appears that the old values are still present so rather than seeing a widget footer I still see footer. I have tested this using only a few widget areas.

Any help is much appreciated.

**Comment by SimpleByDesign** [comment]
I have added a little more regarding the twig runtime error in its own issue [which can be found here](https://github.com/sojimaxi/grav-plugin-widget/issues/8)

**Comment by sojimaxi** [comment]
@SimpleByDesign  I am looking into it

**Comment by SimpleByDesign** [comment]
@sojimaxi Great, thank you.

**Comment by SimpleByDesign** [comment]
Hi,

Thank you for your response. As I am unable to clear down my cache via the `bin/grav clear-cache` command on my server I (I don't have access) have been clearing down the cache folder of all contents manually.

So if I changed the defined widget areas in the theme from

widget:
  areas: [...,footer,...]
to

widget:
  areas: [...,widgetfooter,...]

Then changed / deleted the reference for it as well in the page site.yaml it still remembered the old widget position settings from the theme prior to change.

As I have been manually clearing the cache folder I and not sure if the cmd version clears more than just this area. I have since performed a search an found that a settings file is created under user/config/themes/"theme name".yaml

It was actually this file that kept the original widget location names even after I altered the "theme name".yaml in the theme, deleted any reference in the site.yaml and manually cleared the cache. Of course deleting this file has reset any configuration of the theme, so if anyone has this issue I recommend only deleting from this file anything under and including:

```
widget: 
  area:
    - widgetlocationname
```

Thank you for your assistance in this matter.

Best Wishes
